tests/scripts/parse_notice.py

```diff
@@ -0,0 +1,277 @@
+                        real_package_name: str = self.required_packages[new_package]
+                        print(f"[!] New package found: '{real_package_name}'")
+
+                print("[!] Run the script with the 'fix' argument in order to add it to the NOTICE.txt file")
+                sys.exit(1)
+
+        elif self.mode == "--fix":
+            if real_requirements_name == existing_packages:
+                print("[!] There is no new package listed in the requirements files")
+                sys.exit()
+
+            for new_package in requirements_name_from_file:
+                if self.required_packages[new_package] not in existing_packages:
+                    real_package_name = self.required_packages[new_package]
+
+                    print(f"[!] New package found: '{real_package_name}'")
+                    self.process_package(required_package=new_package)
+                    self.verify_license_in_packages(processed_package=new_package)
+
+                    processed_package = self.processed_packages.get(new_package)
+
+                    if not processed_package:
+                        print(
+                            f"[-] Nothing has been found for package '{real_package_name}' in {self.scanned_json_file}",
+                        )
+                        continue
+
+                    if (
+                        "package_name" not in processed_package
+                        or "version" not in processed_package
+                        or "homepage_url" not in processed_package
+                        or "license_name" not in processed_package
+                        or "license_path" not in processed_package
+                        or "license_content" not in processed_package
+                    ):
+                        print(f"Missing data for '{real_package_name}'. Skipping...")
+                        continue
+
+                    self.write_to_file(processed_package)
+                    print(f"[+] Package '{real_package_name}' has been added to {self.notice_file_name}")
+        else:
+            print("[-] Invalid argument. Please choose an argument between 'fix' or 'check'")
+            sys.exit(1)
+
+    def process_package(self, required_package: str) -> None:
+        """
+        Iterates over the json file outputted by scancode and tries to find a match between
+        the required package and the installed one and looks in the self.POSSIBLE_LICENSE_FILES
+        and self.POSSIBLE_METADATA_FILES where the important information about the package should exist
+        """
+        for entry in self.scanned_results_json["files"]:
+            splitted_entry_path: list[str] = entry["path"].split("/")
+
+            if (
+                splitted_entry_path[-1] in NoticeParser.POSSIBLE_LICENSE_FILES
+                or splitted_entry_path[-1] in NoticeParser.POSSIBLE_METADATA_FILES
+            ):
+                if not len(splitted_entry_path) > 6:
```

Comment:
what's this for?

Reply:
this is found in the `entry["path"]` key: `elastic-serverless-forwarder/venv/lib/python3.9/site-packages/pytest-6.2.5.dist-info/LICENSE`, and I split the path in order to access the name of the package and the LICENSE/METADATA file `['elastic-serverless-forwarder', 'venv', 'lib', 'python3.9', 'site-packages', 'pytest-6.2.5.dist-info', 'LICENSE']`.

That's what I'm checking there, if the LICENSE/METADATA exist in the possible file names and if the length of the splitted is > 6 because it can split paths like `elastic-serverless-forwarder/LICENSE.txt`.